src/middlewared/middlewared/utils/__init__.py

```diff
@@ -442,3 +442,21 @@ def get_service(self, name):
 
     def get_services(self):
         return self._services
+
+# Sync the clock
+def sync_clock():
+    DEFAULT_NTP_SERVER="freebsd.pool.ntp.org"
```

Comment:
@themylogin We should probably use the ntp servers that the end-user has set first and then only use the `DEFAULT_NTP_SERVER` if all else fails. TrueNAS customers block lots of things out of their network.

Reply:
Definitely. Even better if there is some way to query local ntpd for it instead of doing ourselves.

Reply:
Still seems worth to try default NTP servers (we don't have any other on install anyway)

Reply:
@yocalebo @william-gr reminds me about one thing... Can we take advantage of the fact that we can get NTP server via DHCP?

Reply:
We have to retry until it works, the odds are none of this will work right after install

Reply:
> @yocalebo @william-gr reminds me about one thing... Can we take advantage of the fact that we can get NTP server via DHCP?

Yes, option 42 provides NTP but DHCP isn't supported on TN HA systems and we really push paying customers to use static IP addresses on everything.

Reply:
The logic was reimplemented according with feedback.